packages/graph/hash_graph/lib/graph/src/api/rest/utoipa_typedef/subgraph/edges.rs

```diff
@@ -56,7 +77,42 @@ impl From<crate::subgraph::edges::Edges> for Edges {
             knowledge_graph: KnowledgeGraphRootedEdges(edges.knowledge_graph.into_iter().fold(
                 HashMap::new(),
                 |mut map, (id, edges)| {
-                    let edges = edges.into_iter().collect();
+                    let edges = edges.into_iter().map(|edge| {
+                        match edge {
+                            crate::subgraph::edges::KnowledgeGraphOutwardEdges::ToOntology(edge) => {
+                                KnowledgeGraphOutwardEdges::ToOntology(edge)
+                            }
+                            crate::subgraph::edges::KnowledgeGraphOutwardEdges::ToKnowledgeGraph(
+                                edge,
+                            ) => {
+                                let earliest_timestamp = if edge.reversed {
+                                    // We want to log the time the link entity was *first* added
+                                    // from this entity. We therefore need to find the timestamp of
+                                    // the first link entity
```

Comment:
This is no longer true. I think we need to give some reasoning about deduplicating information of edges, where if we have N editions of an entity and M editions of a link with consistent endpoints, we don't want to store N*M edges when 1 is enough.

Reply:
I changed the comment in 14938b548c5dea70a4e51bcd8f3a94cc3e00b5c8